addons/account/wizard/account_payment_register.py

```diff
@@ -308,6 +308,9 @@ def _compute_show_require_partner_bank(self):
     @api.depends('source_amount', 'source_amount_currency', 'source_currency_id', 'company_id', 'currency_id', 'payment_date')
     def _compute_amount(self):
         for wizard in self:
+            if wizard.amount:
+                continue
```

Comment:
Since we don't know which parameter of the compute changed (limitation of the orm), we can't make anything for this.
In multi-currency, the amount must be recomputed when the date changed so this compute must depend of `payment_date`.

I don't understand why you tried to do with such diff but this is really weird... `amount` must always be assigned in the compute method so skipping it with `continue` could lead to a CacheMiss Exception on the orm side. Also, always use the currency when checking a monetary amount.

In brief, there is nothing to fix IMO since this is how work the ORM.

Reply:
The client's issue is that, when you manually change the amount during a payment, then change the date, the amount is reset to the original value.
What I wanted to do was to skip the re-computing the value of the amount, if the amount is already set.